fast_forward/score.py

```diff
@@ -70,19 +71,26 @@ def score_matrix(molname, block, universe, distribution_files, hellinger_weight=
 
     for node1, name1 in block.nodes(data='atomname'):
         for node2, name2 in list(block.nodes(data='atomname'))[node1+1:]:
+            resid1 = block.nodes[node1]['resid']
+            resid2 = block.nodes[node2]['resid']
             atoms = np.array([node1, node2])
-            group_name = f'{name1}_{name2}' # following the naming convention introduced in ITPInteractionMapper
+            group_name = f'{resid1}_{resid2}_{name1}_{name2}' # following the naming convention introduced in ITPInteractionMapper
             indices = find_mol_indices(universe,
                             atoms,
                             molname)
             distr = interaction_distribution(universe, 'distances', indices)
             # calculate simulation distribution
             probs = distr[0].T[1]
-            # read in reference distribution
+            reference_pattern = re.compile(rf'.*(?<!\d){re.escape(group_name)}_distances_distr\.dat$')
```

Comment:
Why do we need the regular expression here? Can't we simply match the exact names? I think partial matching only leads to mistakes that can be avoided.